include/ginkgo/core/factorization/ilu.hpp

```diff
@@ -25,6 +25,14 @@ namespace gko {
 namespace factorization {
 
 
+/**
+ * A helper for algorithm selection in the incomplete factorization.
+ * sparselib is only available for cuda and hip.
+ * syncfree is Ginkgo's implementation through the Lu factorization with given
+ * sparsity.
+ */
+enum class factorize_algorithm { sparselib, syncfree };
```

Comment:
I will also use it to IC, so it will be for incomplete factorize algorithm at least.
maybe `incomplete_factorize_algorithm` if the original one is too general?

Reply:
That makes sense. I'm mostly thinking about code readability/logicality on the user side. I do think it would be good to specify that this is only for incomplete somehow, though `incomplete_factorize_algorithm` is pretty long. What if we had the same enum for both ilu and ic but with different names (`ilu_algorithm` and `ic_algorithm`)? But we can also go with the longer name. Maybe @upsj has thoughts on this?